packages/back-end/src/models/EventModel.ts

```diff
@@ -0,0 +1,111 @@
+import z from "zod";
+import _ from "lodash";
+import mongoose from "mongoose";
+import {
+  NotificationEventName,
+  notificationEventNames,
+  NotificationEventPayload,
+  NotificationEventResource,
+  notificationEventResources,
+} from "../events/base-types";
+import { EventInterface } from "../../types/event";
+import { errorStringFromZodResult } from "../util/validation";
+
+const eventSchema = new mongoose.Schema({
+  id: {
+    type: String,
+    unique: true,
+    required: true,
+  },
+  dateCreated: Date,
+  organizationId: {
+    type: String,
+    required: true,
+  },
+  data: {
+    type: Object,
+    required: true,
+    validate: {
+      validator(value: unknown) {
+        // NotificationEventPayload<EventName, ResourceType, DataType>
+        const zodSchema = z
+          .object({
+            organization_id: z.string().optional(),
+            event_id: z.string().startsWith("event-"),
+            event: z.enum(notificationEventNames),
+            object: z.enum(notificationEventResources),
+            data: z.any(),
+          })
+          .strict();
+
+        const result = zodSchema.safeParse(value);
+
+        if (!result.success) {
+          const errorString = errorStringFromZodResult(result);
+          console.error("Invalid Event data ", errorString);
```

Comment:
Instead of `console.error`, you can 

```ts
import { logger } from "../util/logger";
logger.error(msg, context);
```

That will add additional context and format the error so it can be parsed correctly by our production log monitoring.